# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/push/PushManager.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 dom/push/PushManager.cpp
--- a/dom/push/PushManager.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/dom/push/PushManager.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -102,7 +102,7 @@
                                 nsTArray<uint8_t>&& aRawP256dhKey,
                                 nsTArray<uint8_t>&& aAuthSecret,
                                 nsTArray<uint8_t>&& aAppServerKey)
-      : WorkerThreadRunnable(aWorkerPrivate, "GetSubscriptionResultRunnable"),
+      : WorkerThreadRunnable("GetSubscriptionResultRunnable"),
         mProxy(std::move(aProxy)),
         mStatus(aStatus),
         mEndpoint(aEndpoint),
@@ -183,7 +183,7 @@
         worker, std::move(mProxy), aStatus, endpoint, mScope,
         std::move(mExpirationTime), std::move(rawP256dhKey),
         std::move(authSecret), std::move(appServerKey));
-    if (!r->Dispatch()) {
+    if (!r->Dispatch(worker)) {
       return NS_ERROR_UNEXPECTED;
     }
 
@@ -296,8 +296,7 @@
  public:
   PermissionResultRunnable(PromiseWorkerProxy* aProxy, nsresult aStatus,
                            PermissionState aState)
-      : WorkerThreadRunnable(aProxy->GetWorkerPrivate(),
-                             "PermissionResultRunnable"),
+      : WorkerThreadRunnable("PermissionResultRunnable"),
         mProxy(aProxy),
         mStatus(aStatus),
         mState(aState) {
@@ -352,7 +351,7 @@
 
     // This can fail if the worker thread is already shutting down, but there's
     // nothing we can do in that case.
-    Unused << NS_WARN_IF(!r->Dispatch());
+    Unused << NS_WARN_IF(!r->Dispatch(mProxy->GetWorkerPrivate()));
 
     return NS_OK;
   }